core/includes/common.inc

```diff
@@ -7494,20 +7683,35 @@ function backdrop_sort(array &$array, array $keys = array('weight'), $dir = SORT
 
 /**
  * Checks if the key is a property.
+ *
+ * @param string $key
+ *   The key.
+ *
+ * @return bool
  */
 function element_property($key) {
   return substr($key, 0, 1) == '#';
 }
 
 /**
  * Gets properties of a structured array element (keys beginning with '#').
+ *
+ * @param array $element
+ *   The element array whose properties are to be identified.
+ *
+ * @return array
  */
-function element_properties($element) {
+function element_properties(array $element) {
   return array_filter(array_keys((array) $element), 'element_property');
```

Comment:
The mix of type hint and casting (one line after) makes me wonder, how to deal with this ancient code.

Sure, it's *supposed* to be an array, but apparently it isn't or wasn't always. (?)
For sure the casting could go, if we type hint the param. Something other than array wouldn't even get there.
But it's really old code, almost 20 years it seems... 🤔